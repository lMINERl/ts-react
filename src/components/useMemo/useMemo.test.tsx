import React, { useMemo } from 'react';
import { shallow } from '../../test/Configurations';
import UseMemo from './useMemo';
import { render, getByTestId, fireEvent } from '@testing-library/react';

describe('<UseMemo />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<UseMemo initiate={true} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('has number display', () => {
    const { container } = render(<UseMemo initiate={true} />);
    const renderCount = getByTestId(container, 'number-display');
    expect(renderCount.textContent).toMatch(/[a-zA-Z]* \: [0-9]+/);
  });

  it('has button', () => {
    const { container } = render(<UseMemo initiate={true} />);
    const renderCount = getByTestId(container, 'incremental-button');
    expect(renderCount.nodeName).toBe('BUTTON');
  });

  it('should increase count from initial 0 to 1', () => {
    const { container } = render(<UseMemo initiate={true} />);
    const incrementButton = getByTestId(container, 'incremental-button');
    const displayCount = getByTestId(container, 'number-display');
    fireEvent.click(incrementButton);
    expect(displayCount.textContent).toMatch(/[a-zA-Z]* \: 1/);
  });
});
