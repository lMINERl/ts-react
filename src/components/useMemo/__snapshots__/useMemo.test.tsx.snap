// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseMemo /> renders without crashing 1`] = `"<div class=\\"App\\"><header class=\\"App-header\\"><h3>UseMemo</h3><h2 data-testid=\\"number-display\\">i : 0</h2><div data-testid=\\"render-count\\">render count : 0</div><div> <div data-testid=\\"render-count\\">render count : 0</div> </div><button data-testid=\\"incremental-button\\">Increment I</button></header></div>"`;
